File: binary_gap/binary_gap.rb
def solution(n)
  n_array = n.to_s(2).split('')
  
  i = 0 
  max_length = 0
  current_length = 0
  while i < n_array.length
    
    if n_array[i] == '1'
      if current_length >  max_length
        max_length = current_length
      end
      
      current_length = 0
    else
      current_length += 1
    end      
    i += 1  
  end
  
  max_length  
end

File: binary_gap/binary_gap.js
function solution(N) {
  var arr = N.toString(2).split("");
  var length = 0;
  var maxLength = 0;
  for(var i = 0; i < arr.length; i++){
    if(arr[i] === "1"){
      if(length > maxLength){
        maxLength = length;
      }
      length = 0;
    }else{
      length += 1;
    }
  }
  return maxlength;
}